kaldb/src/main/java/com/slack/kaldb/server/ManagerApiGrpc.java

```diff
@@ -171,6 +180,29 @@ public void updatePartitionAssignment(
     }
   }
 
+  @Override
+  public void restoreSnapshot(
+      ManagerApi.RestoreSnapshotRequest request,
+      StreamObserver<ManagerApi.RestoreSnapshotResponse> responseObserver) {
+    List<SnapshotMetadata> snapshotsToRestore =
```

Comment:
You may want to add some validation here around the request - ie, what happens if you have a GRPC request with nonsense data? Like an end time before a start time? Or an empty service name?